src/main/scala/dependencies/ModelFactory.scala

```diff
@@ -52,24 +52,24 @@ object ModelFactory {
       * Take this opportunity to also squash dependencies for the same artifact at different versions
       * down to a single version (the highest).
       */
-    val thirdParty: Set[MavenDependency] = {
-      def flattenDependencies(root: MavenDependency): Seq[MavenDependency] =
+    val thirdParty: Set[MavenDependency[MavenArtifactCoordinates]] = {
+      def flattenDependencies(root: MavenDependency[ArtifactCoordinates]): Seq[MavenDependency[MavenArtifactCoordinates]] =
         if (root.coordinates.isInstanceOf[MavenArtifactCoordinates]) {
-          Seq(root) ++ root.dependsOn.toSeq.flatMap(flattenDependencies)
+          Seq(root.asInstanceOf[MavenDependency[MavenArtifactCoordinates]]) ++ root.dependsOn.toSeq.flatMap(flattenDependencies)
         } else {
           root.dependsOn.toSeq.flatMap(flattenDependencies)
         }
-
+      
       roots.flatMap(t => flattenDependencies(t._2))
-          .groupBy(d => d.coordinates match {
-            case m: MavenArtifactCoordinates => m.artifact
-            case _ => throw new IllegalStateException("Should not see non-maven coordinates here: " + d.coordinates)
-          })
+          .groupBy(_.coordinates.artifact)
```

Comment:
The key is that `MavenAccountCoordinates` have both an `artifact` (e.g. `some.group.id:some.artifact.id`), and a version.  Group by the artifact, then we can pluck the max version as needed, below.